gui/src/installer/step/descriptor.rs

```diff
@@ -1399,6 +1401,11 @@ impl Step for RegisterDescriptor {
         for key in ctx.keys.iter().filter(|k| !k.name.is_empty()) {
             map.insert(key.master_fingerprint, key.name.clone());
         }
+        // we reset device registered set if the descriptor have changed
+        if self.registered_descriptor != self.descriptor {
+            self.registered = Default::default();
+            self.registered_descriptor.clone_from(&self.descriptor);
+        }
```

Comment:
I think we do not need to have a new field. example:
```rust
fn load_context(&mut self, ctx: &Context) {
        // we reset device registered set if the descriptor have changed
        if self.descriptor != ctx.descriptor {
            self.registered = Default::default();
        }
        self.descriptor.clone_from(&ctx.descriptor);
        let mut map = HashMap::new();
        for key in ctx.keys.iter().filter(|k| !k.name.is_empty()) {
            map.insert(key.master_fingerprint, key.name.clone());
        }
     }
```

Reply:
adressed